examples/MomentoRedisLambdaExample/MomentoRedisExampleLambdaHandler/src/MomentoRedisExampleLambdaHandler/MomentoCache.cs

```diff
@@ -0,0 +1,188 @@
+using System.Diagnostics;
+using Momento.Sdk;
+using Momento.Sdk.Responses;
+
+namespace MomentoRedisExampleLambdaHandler;
+
+public class MomentoCache
+{
+    private string _cacheName;
+    private CacheClient _client;
+
+    public MomentoCache(CacheClient client, string cacheName)
+    {
+        _client = client;
+        _cacheName = cacheName;
+    }
+    
+    public async Task CreateMomentoCache()
+    {
+        var response = await _client.CreateCacheAsync(_cacheName);
+
+        switch (response)
+        {
+            case CreateCacheResponse.Success:
+                Console.WriteLine($"\t Cache: {_cacheName} created successfully.");
+                break;
+            case CreateCacheResponse.CacheAlreadyExists:
+                Console.WriteLine($"\t Cache: {_cacheName} already exists.");
+                break;
+            case CreateCacheResponse.Error:
+                Console.WriteLine($"\t Error creating cache: {_cacheName}");
+                break;
+        }
+    }
+    
+    public async Task<List<Task<SetResult>>> DoMomentoSets(int numMomentoSetRequests, bool withSleep = false)
+    {
+        Console.WriteLine($"\t WITH SLEEP flag: {withSleep}");
+        try
+        {
+            List<Task<SetResult>> momentoSetTasks = new List<Task<SetResult>>();
+            string sourceData = await Utils.GetSourceData();
```

Comment:
this looks to me like the only place where you are using an await in this function, meaning that if you didn't have to do this here, this function could be synchronous, which would make the calling code a little easier to understand.

What I would do is make this sourceData variable be an argument to the function. Then, you can acquire it once at the beginning of the main function (including the await that you need to do in order to realize it), and pass the string reference into this function.